### PR TITLE
add metrics, update postman

### PR DESCRIPTION
**Pull request by prekoa**

### CODE REVIEW

File: cluster/manager.go [hunk 1]
```diff
@@ -50,6 +52,13 @@ func NewManager(clusters clusterRepository, secrets secretValidator, events clus
 		clusters: clusters,
 		secrets:  secrets,
 		events:   events,
+		statusChangeDuration: prometheus.NewSummaryVec(prometheus.SummaryOpts{
```
Comment (sagikazarmark) [inline cluster/manager.go]: Create the collector outside and inject it into the manager.

File: cluster/manager.go [hunk 2]
```diff
@@ -63,3 +72,7 @@ func (m *Manager) getLogger(ctx context.Context) logrus.FieldLogger {
 func (m *Manager) getErrorHandler(ctx context.Context) emperror.Handler {
 	return pipelineContext.ErrorHandlerWithCorrelationID(ctx, m.errorHandler)
 }
+
+func (m *Manager) GetCollector() prometheus.Collector {
```
Comment (sagikazarmark) [inline cluster/manager.go]: When injected, this method becomes unnecessary.

File: internal/platform/gin/drain.go [hunk 3]
```diff
@@ -22,22 +22,29 @@ import (
...
+	mu        sync.RWMutex
+	drainMode prometheus.Gauge
 
 	errorHandler emperror.Handler
 }
 
 // NewDrainModeMiddleware returns a new DrainModeMiddleware instance.
 func NewDrainModeMiddleware(errorHandler emperror.Handler) *DrainModeMiddleware {
 	return &DrainModeMiddleware{
+		drainMode: prometheus.NewGauge(prometheus.GaugeOpts{
```
Comment (sagikazarmark) [inline internal/platform/gin/drain.go]: inject the metric instead

File: internal/platform/gin/drain.go [hunk 4]
```diff
@@ -22,22 +22,29 @@ import (
...
 
 var basePath = viper.GetString("pipeline.basepath")
 
 // DrainModeMiddleware prevents write operations from succeeding.
 type DrainModeMiddleware struct {
-	enabled bool
-	mu      sync.RWMutex
+	enabled   bool
+	mu        sync.RWMutex
+	drainMode prometheus.Gauge
```
Comment (sagikazarmark) [inline internal/platform/gin/drain.go]: Call it maybe drainModeMetric or drainModeGauge or something that tells us what this is.

File: internal/platform/gin/drain.go [hunk 5]
```diff
@@ -110,3 +119,7 @@ func isException(c *gin.Context) bool {
 
 	return false
 }
+
+func (m *DrainModeMiddleware) GetCollector() prometheus.Collector {
```
Comment (sagikazarmark) [inline internal/platform/gin/drain.go]: When injected, this becomes unnecessary.

File: cluster/manager.go (hunk 1 above)
Comment (prekoa) [reply cluster/manager.go]: fixed

File: internal/platform/gin/drain.go (hunk 3 above)
Comment (prekoa) [reply internal/platform/gin/drain.go]: fixed, thx

File: cmd/pipeline/main.go [hunk 6]
```diff
@@ -59,14 +59,28 @@ import (
 var log *logrus.Logger
 var logger *logrus.Entry
 
+var statusChangeDuration = prometheus.NewSummaryVec(prometheus.SummaryOpts{
```
Comment (sagikazarmark) [inline cmd/pipeline/main.go]: Do not create this as global variables

File: cluster/manager_delete.go [hunk 7]
```diff
@@ -45,10 +46,10 @@ func (m *Manager) DeleteCluster(ctx context.Context, cluster CommonCluster, forc
 		return err
 	}
 	var timer *prometheus.Timer
-	if viper.GetBool("metrics.debug") {
-		timer = prometheus.NewTimer(StatusChangeDuration.WithLabelValues(cluster.GetCloud(), cluster.GetLocation(), pkgCluster.Deleting, org.Name, cluster.GetName()))
+	if viper.GetBool(config.MetricsDebug) {
```
Comment (stoader) [inline cluster/manager_delete.go]: Factor out the creation of prometheus timer into a function of `Manager`. This function can be used from manager_delete.go as well

Comment (prekoa) [reply cluster/manager_delete.go]: thx, fixed

**Pull request merged**